bitcoin/tx.c

```diff
@@ -324,6 +325,16 @@ const u8 *bitcoin_tx_output_get_script(const tal_t *ctx,
 	return cln_wally_tx_output_get_script(ctx, output);
 }
 
+bool bitcoin_tx_output_script_is_p2wsh(const struct bitcoin_tx *tx, int outnum)
+{	const struct wally_tx_output *output;
+	assert(outnum < tx->wtx->num_outputs);
+	output = &tx->wtx->outputs[outnum];
+
+	return output->script_len == BITCOIN_SCRIPTPUBKEY_P2WSH_LEN &&
+	       output->script[0] == OP_0 &&
+	       output->script[1] != OP_PUSHBYTES(sizeof(struct ripemd160));
```

Comment:
This should be the sizeof a sha256 struct, not ripemd160 I believe:

- `OP_0 [20 byte hash160]` -> p2wpkh
- `OP_O [32 byte sha256]` -> p2wsh

Reply:
Thanks @jgriffiths, this actually got the performance up again.